<a href="https://colab.research.google.com/github/twumasimb/Exploratory-Data-Analysis/blob/main/EDA_final.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [22]:
# Importing necessary libraries
import matplotlib.pyplot as plt
import numpy as np
import seaborn as sns
import pandas as pd
import statsmodels.api as sm
from scipy.stats import norm
import random

In [3]:
data = pd.read_csv('/content/sample_data/dataset.csv') # Importing the dataset

In [16]:
# Population statistics
data.describe()

ptr      TOT_ENR_M      TOT_ENR_F        TOT_ENR  Mean_rating
count  1020.000000    1020.000000    1020.000000    1020.000000  1020.000000
mean     12.724380    2545.966667    2409.072549    4955.039216     3.204280
std       2.493322    7152.729136    6819.987053   13972.312388     0.539454
min       4.070981      10.000000      10.000000      20.000000     1.750000
25%      11.012754     195.750000     190.500000     385.500000     2.875000
50%      12.769690     509.500000     460.500000     971.500000     3.250000
75%      14.688600    1611.250000    1512.500000    3104.250000     3.625000
max      21.200001  109666.000000  106323.000000  215989.000000     4.000000

In [5]:
#randomly selecting 100 datapoints
random.seed(20) # Ensure we get the same results all the time

sampled_data = data.sample(n=100)


In [8]:
#peeping at sampled data
sampled_data.describe()

ptr     TOT_ENR_M     TOT_ENR_F       TOT_ENR  Mean_rating
count  100.000000    100.000000    100.000000    100.000000   100.000000
mean    12.676574   2069.820000   1941.280000   4011.100000     3.217679
std      2.385016   5191.135238   4877.413197  10068.382474     0.547758
min      7.000000     33.000000     30.000000     77.000000     2.000000
25%     10.972796    212.250000    196.500000    403.750000     2.875000
50%     12.874595    525.000000    507.000000   1025.500000     3.250000
75%     14.352116   1448.500000   1334.250000   2799.750000     3.750000
max     17.141464  29699.000000  27653.000000  57352.000000     4.000000

In [13]:
dataset = sampled_data.drop(['TOT_ENR_M', 'TOT_ENR_F', 'TOT_ENR'], axis=1)

In [15]:
dataset.describe() # Sampled data statistics

ptr  Mean_rating
count  100.000000   100.000000
mean    12.676574     3.217679
std      2.385016     0.547758
min      7.000000     2.000000
25%     10.972796     2.875000
50%     12.874595     3.250000
75%     14.352116     3.750000
max     17.141464     4.000000

**Making Statistical Claims**



1.   50% of the ISDs have a student to teacher ratio above 12
2.   The mean student to teacher ratio is at least 12.50



1. ***Sample Proportion***

H0: p = 0.5\
H0: p!= 0.5

In [19]:
# Count number of ISD with ptr > 12
count = (dataset.ptr > 12).sum().sum()

In [18]:
count

62

There are 62 ISD with ptr greater than 12. 

Therefore, the sample proportion p_hat = 0.62

In [21]:
# set the sample size and number of successes
n = 100
p = 0.5
x = 62

# calculate the sample proportion
sample_prop = x / n

# calculate the population proportion
pop_prop = p

# calculate the standard error
se = np.sqrt((pop_prop * (1 - pop_prop)) / n)

# calculate the z-score
z_score = (sample_prop - pop_prop) / se

# calculate the p-value (two tailed hence the factor of 2)
p_value = 2 * (1 - norm.cdf(abs(z_score)))

# print the results
print('z-score:', z_score)
print('p-value:', p_value)


z-score: 2.4
p-value: 0.01639507184919231


The test statistics are\
z-score: 2.4\
p-value: 0.01639507184919231\
Hence it is safe to say that we reject the null hypothesis since the p-value is far less than the significant level of alpha = 0.05 or 95%

2. ***Sample Mean***

H0: mean = 12.50\
H1: mean < 12.50\

Known data about the population\
population mean = 12.72\
population std = 2.49\
sample std = 2.39

In [25]:
# set the population mean and standard deviation
pop_mean = 12.72
pop_std = 2.49
n = 100
sample_mean = 12.50

# calculate the z-score
z_score = (sample_mean - pop_mean) / (pop_std / np.sqrt(n))

# calculate the p-value (The value is left tailed)
p_value = (1 - norm.cdf(abs(z_score)))

# print the results
print('z-score:', z_score)
print('p-value:', p_value)


z-score: -0.8835341365461872
p-value: 0.18847387353552802


The test statistics are \
z-score: -0.8835341365461872\
p-value: 0.18847387353552802

Hence we fail to reject the null hypothesis since p-value > alpha = 0.05 (95%)